dotnet/src/Connectors/Connectors.AzureOpenAI/Settings/AzureOpenAIAudioToTextExecutionSettings.cs

```diff
@@ -0,0 +1,221 @@
+    /// </summary>
+    public AzureOpenAIAudioToTextExecutionSettings()
+        : this(DefaultFilename)
+    {
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="AzureOpenAIAudioToTextExecutionSettings"/> class.
+    /// </summary>
+    /// <param name="filename">Filename or identifier associated with audio data. Should be in format {filename}.{extension}</param>
+    public AzureOpenAIAudioToTextExecutionSettings(string filename)
+    {
+        this._filename = filename;
+    }
+
+    /// <inheritdoc/>
+    public override PromptExecutionSettings Clone()
+    {
+        return new AzureOpenAIAudioToTextExecutionSettings(this.Filename)
+        {
+            ModelId = this.ModelId,
+            ExtensionData = this.ExtensionData is not null ? new Dictionary<string, object>(this.ExtensionData) : null,
+            Temperature = this.Temperature,
+            ResponseFormat = this.ResponseFormat,
+            Language = this.Language,
+            Prompt = this.Prompt
+        };
+    }
+
+    /// <summary>
+    /// Converts <see cref="PromptExecutionSettings"/> to derived <see cref="AzureOpenAIAudioToTextExecutionSettings"/> type.
+    /// </summary>
+    /// <param name="executionSettings">Instance of <see cref="PromptExecutionSettings"/>.</param>
+    /// <returns>Instance of <see cref="AzureOpenAIAudioToTextExecutionSettings"/>.</returns>
+    public static AzureOpenAIAudioToTextExecutionSettings FromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        if (executionSettings is null)
+        {
+            return new AzureOpenAIAudioToTextExecutionSettings();
+        }
+
+        if (executionSettings is AzureOpenAIAudioToTextExecutionSettings settings)
+        {
+            return settings;
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var openAIExecutionSettings = JsonSerializer.Deserialize<AzureOpenAIAudioToTextExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
+
+        if (openAIExecutionSettings is not null)
+        {
+            return openAIExecutionSettings;
+        }
+
+        throw new ArgumentException($"Invalid execution settings, cannot convert to {nameof(AzureOpenAIAudioToTextExecutionSettings)}", nameof(executionSettings));
+    }
+
+    /// <summary>
+    /// The timestamp granularities available to populate transcriptions.
+    /// </summary>
+    public enum TimeStampGranularities
+    {
+        /// <summary>
+        /// Not specified.
+        /// </summary>
+        Default = 0,
+
+        /// <summary>
+        /// The transcription is segmented by word.
+        /// </summary>
+        Word = 1,
+
+        /// <summary>
+        /// The timestamp of transcription is by segment.
+        /// </summary>
+        Segment = 2,
+    }
+
+    /// <summary>
+    /// Specifies the format of the audio transcription.
+    /// </summary>
+    public enum AudioTranscriptionFormat
```

Comment:
Right, there was no format before. However, as discussed offline, it was added because the underlying library changed the ResponseFormat property type from a string to an enum. It's one of the breaking changes that are tracked in this issue: https://github.com/microsoft/semantic-kernel/issues/7053. This will be discussed later to decide which changes to keep and which to avoid.